src/dns/nm-dns-dnsmasq.c

```diff
@@ -356,22 +354,16 @@ start_dnsmasq (NMDnsDnsmasq *self)
 		return;
 	}
 
-	dbus_mgr = nm_dbus_manager_get ();
-	g_return_if_fail (dbus_mgr);
-
-	connection = nm_dbus_manager_get_connection (dbus_mgr);
-	g_return_if_fail (connection);
-
 	priv->dnsmasq_cancellable = g_cancellable_new ();
-	g_dbus_proxy_new (connection,
-	                  G_DBUS_PROXY_FLAGS_DO_NOT_AUTO_START,
-	                  NULL,
-	                  DNSMASQ_DBUS_SERVICE,
-	                  DNSMASQ_DBUS_PATH,
-	                  DNSMASQ_DBUS_SERVICE,
-	                  priv->dnsmasq_cancellable,
-	                  dnsmasq_proxy_cb,
-	                  self);
+	g_dbus_proxy_new_for_bus (G_BUS_TYPE_SYSTEM,
```

Comment:
why is this better? Previously, at least `NMDBusManager` was the authority regarding our D-Bus session.

Ok, we use `G_BUS_TYPE_SYSTEM` **all** over the place. So, in that spirit, the patch is certainly correct, and the commit message could just say: "like we do at ~30 other places."

I still think `NMDBusManager` should have a say which D-Bus connection to use, but anyway, the patch doesn't change that.